app/modules/submissions/actions/submission-clone-all.js

```diff
@@ -2,15 +2,16 @@ import { selected } from "../selectors"
 import { submissionCloneFunc } from "./submission-clone"
 
 import { clone } from "../../../lib/cloneutils"
+import Promise from "bluebird"
 
 const submissionClone = submissionCloneFunc(clone)
-
 // PUBLIC: Async thunk action for cloning all selected submissions.
 export const submissionCloneAll = () => {
   return (dispatch, getState) => {
-    const selectedSubmissions = selected(getState())
-    const promises = []
-    selectedSubmissions.forEach((each) => promises.push(dispatch(submissionClone(each))))
-    return Promise.all(promises).then()
+    var selectedSubmissions = selected(getState())
+    return Promise.map(selectedSubmissions, submission => {
+      return dispatch(submissionClone(submission))
+    },
+    {concurrency: 2}).then()
```

Comment:
Limit clone concurrency to two repos at a time, can probably be increased when we switch to electron bindings for git.

Reply:
Can this lonely `then()`  be dropped, or is there some quirk with `bluebird` that requires this?